src/medley/core.cljc

```diff
@@ -506,6 +506,42 @@
   ([pred coll]
    (partition-between (fn [_ x] (pred x)) coll)))
 
+(defn window
+  "A sliding window, returning a lazy sequence of partitions, containing each
+  element and n-1 preceeding elements, when present. Therefore partitions at the
+  start may contain fewer items than the rest. Returns a stateful transducer
+  when no collection is provided. For a sliding window containing each element
+  and n-1 _following_ elements, use `clojure.core/partition` with a `step` size
+  of 1."
+  {:added "1.10.0"}
+  ([n]
+   (fn [rf]
+     (if (zero? n)
+       (fn
+         ([] (rf))
+         ([result] (unreduced (rf result)))
+         ([result _x] (rf result [])))
+       (let [part #?(:clj (java.util.ArrayList. n) :cljs (array))]
+         (fn
+           ([] (rf))
+           ([result]
+            (unreduced (rf result)))
+           ([result x]
+            (when (= n #?(:clj (.size part) :cljs (.-length part)))
+              #?(:clj (.remove part 0) :cljs (.shift part)))
+            (when-not (reduced? result)
+              #?(:clj (.add part x) :cljs (.push part x)))
+            (rf result (vec #?(:clj (.toArray part) :cljs (.slice part))))))))))
+  ([n coll]
+   (letfn [(part [part-n coll]
+             (let [run (doall (take part-n coll))]
+               (lazy-seq
+                (when (== part-n (count run))
```

Comment:
Is using `==` over `=` due to performance reasons? What do the benchmarks come out with when comparing the two?

Reply:
Yeah, performance hunch. Not particularly thought out. Will run in through criterium, but not expecting a meaningful difference.

Reply:
```clojure
(c/benchmark-round-robin [(window== 1000 (range 1e8)) (window= 1000 (range 1e8))] c/*default-benchmark-opts*)
;; window== sample mean: 3.3533791024410264E-7
;; window= sample mean:  3.5284077743305506E-7
```

so a ~17.5ns or 5% difference

expect less difference for small collections

Reply:
```clojure
(c/benchmark-round-robin [(dorun (window== 1000 (range 10000))) (dorun (window= 1000 (range 10000)))] c/*default-benchmark-opts*)
;; window== sample mean: 0.29017708853649266
;; window= sample mean:  0.3584622250301741

;; window== mean: 0.3225724956375 
;; window= mean:  0.3290516003666667
```

my stats isn't strong enough to be sure sample mean is a better indicator than mean, so i'll leave both means for you to look at. either a ~20% difference or a 2% difference. but both in favour of `==` over `=` so probably worth keeping, but I'm easy.

Reply:
from my reading, sample mean will tend to emphasise the impact of outliers, thereby highlighting variance in one direction or another. so in this case, larger sample-mean draws attention to the fact a few of the test runs show significantly slower runs for =